### ISSUE TITLE
parsing failure

### ISSUE THREAD

**Issue by kmcdonell** [koalaman/shellcheck#3400]
#### For bugs with existing features

- Rule Id: SC1073
- My shellcheck version: 0.9.0
- [yes] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
for a in -d -f
do
        [ $a foo ] && echo bar
done
```

#### Here's what shellcheck currently says:
In eek line 4:
        [ $a foo ] && echo bar
        ^-- SC1073 (error): Couldn't parse this test expression. Fix to allow more checks.
             ^-- SC1072 (error): Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:
Nothing, this is valid shell syntax.
